src/main/java/org/wildfly/security/auth/realm/ldap/UserPasswordCredentialLoader.java

```diff
@@ -113,8 +117,9 @@ public SupportLevel getCredentialAcquireSupport(final Class<? extends Credential
 
         @Override
         public SupportLevel getEvidenceVerifySupport(final Class<? extends Evidence> evidenceType, final String algorithmName) throws RealmUnavailableException {
-            // If it is obtainable, it is verifiable.
-            return getCredentialAcquireSupport(PasswordCredential.class, null);
+            // If we can acquire PasswordCredential and it suppport provided evidence, we can verify.
+            if ( ! PasswordCredential.canVerifyEvidence(evidenceType, algorithmName)) return SupportLevel.UNSUPPORTED;
+            return getCredentialAcquireSupport(PasswordCredential.class, algorithmName);
```

Comment:
This code is misleading, because ``PasswordCredential.canVerifyEvidence`` always returns ``false`` when ``algorithmName`` is not ``null``.  This fact hides the fact that ``PasswordGuessEvidence`` does not accept algorithm names, because line 122 here is only reached when ``algorithmName`` is ``null``.  At a minimum, I think it is best to always explicitly pass ``null`` to ``getCredentialAcquireSupport`` to make it clear that any password credential algorithm is allowed.